src/components/cc-product-list/products-controller.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * @typedef {import('./cc-product-list.types.js').ProductSection} ProductSection
+ * @typedef {import('./cc-product-list').CcProductList} CcProductList
+ * @typedef {import('./cc-product-list.types.js').Category} Category
+ */
+
+export class ProductsController {
+
+  constructor (host) {
+    /** @type {CcProductList} The host. */
+    this._host = host;
+
+    /** @type {ProductSection[]} The initial product list. */
+    this._productsList = [];
+
+    /** @type {ProductSection[]} product list filtered by the category selected nor with the input */
+    this._currentProductsList = [];
+
+    /** @type {Category[]}  */
+    this._categories = [];
+
+    /** @type {'all'|string}  */
+    this._currentCategory = 'all';
+  }
+
+  getProductsList () {
+    return this._currentProductsList;
+  }
+
+  /**
+   * @param {ProductSection[]} productsList
+   */
+  set productsList (productsList) {
```

Comment:
No, we don't have to do that. If the component gets a new `productsList`, the component will have to rerender. Before rendering again it will go in the `willUpdate` and the controller will be updated with the new `productsList` so there's no need to trigger an update again.